app/forms/incidents/told_on_form.rb

```diff
@@ -1,69 +1,84 @@
     include BaseForm
-
     form_for Incident
 
     attr_accessor :told_year, :told_month, :told_day
     attr_writer :told_on
 
-    validates :told_on, presence: true, unless: :draft_and_not_incomplete_date?
+    attr_accessor :occurred_year, :occurred_month, :occurred_day
+    attr_writer :occurred_on
+
+    validates :told_on, presence: true, unless: :draft_and_not_partially_complete_date?
     validates :told_on, date: { not_in_the_future: true }, allow_nil: true
 
+    validates :occurred_on, presence: true, unless: :draft_and_not_partially_complete_date?
+    validates :occurred_on, date: { not_in_the_future: true }, allow_nil: true
+
+    validate :occurred_on_before_told_on
+
     def initialize(*args)
       super
       set_instance_variables_for_attributes_if_not_set_but_in_model(
-        attrs: told_on_fields,
-        model_attributes: told_on_from_model
+        attrs: (told_on_date_fields.fields + occurred_on_date_fields.fields),
+        model_attributes:
+        [
+          told_on_date_fields.model_attributes,
+          occurred_on_date_fields.model_attributes
+        ].compact.reduce(&:merge)
       )
     end
 
     def told_on
       return @told_on if @told_on.present?
-      return incomplete_date if told_on_methods.any?(&:blank?)
+      return if told_on_date_fields.blank?
+      return :invalid if told_on_date_fields.partially_complete? || told_on_date_fields.form_date_invalid?
 
-      @told_on = attributes[:told_on] = Date.new(*told_on_methods.map(&:to_i))
-    rescue ArgumentError
-      :invalid
+      @told_on = attributes[:told_on] = told_on_date_fields.form_date
     end
 
-    private
-
-    def exclude_from_model
-      told_on_fields
-    end
+    def occurred_on
+      return @occurred_on if @occurred_on.present?
+      return if occurred_on_date_fields.blank?
+      return :invalid if occurred_on_date_fields.partially_complete? || occurred_on_date_fields.form_date_invalid?
 
-    def told_on_fields
-      %i[told_year told_month told_day]
+      @occurred_on = attributes[:occurred_on] = occurred_on_date_fields.form_date
     end
 
-    def told_on_methods
-      @told_on_methods ||= told_on_fields.map { |f| __send__(f) }
-    end
+    private
 
-    def told_on_from_model
-      return unless model.told_on?
+    def occurred_on_before_told_on
+      return if occurred_on.blank? || told_on.blank?
 
-      {
-        told_year: model.told_on.year,
-        told_month: model.told_on.month,
-        told_day: model.told_on.day
-      }
+      errors.add(:occurred_on, :invalid_timeline) if told_on < occurred_on
+    rescue ArgumentError
+      :invalid
     end
 
-    def incomplete_date
-      return nil if told_on_methods.all?(&:blank?)
+    def exclude_from_model
+      (told_on_date_fields.fields + occurred_on_date_fields.fields)
+    end
 
-      @incomplete_date = true
-      :invalid
+    def draft_and_not_partially_complete_date?
+      draft? && !told_on_date_fields.partially_complete?
```

Comment:
This only checks whether `told_on_date_fields` are partially complete. It will not detect if `occurred_on_date_fields` are partially complete. A simple solution would be:
```ruby
draft? && !told_on_date_fields.partially_complete? && occurred_on_date_fields.partially_complete?
```
However, that could cause an incorrect error if one date was empty, the other partially complete, and draft selected. In this case both sets of date fields would get an error - were the correct behaviour would be just for the incomplete date fields to get the error.

So, I think this is a better solution:

```
def draft_and_not_partially_complete_told_on_date?
  draft? && !told_on_date_fields.partially_complete?
end

def draft_and_not_partially_complete_occurred_on_date?
  draft? && !occurred_on_date_fields.partially_complete?
end
```
Then use `draft_and_not_partially_complete_told_on_date?` in the `unless` clause of the `told_on` validation, and the other method with the `occurred_on` validation.